lib/AST/ASTPrinter.cpp

```diff
@@ -137,6 +137,11 @@ static bool contributesToParentTypeStorage(const AbstractStorageDecl *ASD) {
   return !ND->isResilient() && ASD->hasStorage() && !ASD->isStatic();
 }
 
+static bool isInObjCImpl(const ValueDecl *VD) {
+  auto *ED = dyn_cast<ExtensionDecl>(VD->getDeclContext());
```

Comment:
I believe it only needs to apply to direct members—if you somehow declared, say, a nested type inside an `@_objcImplementation` extension (which would be pretty weird in the first place), you wouldn't want to apply special logic to its members. Recall that accessors are represented in the AST as peers of their storage, not members of their storage, and that one extension cannot be nested inside another. If you think there's a case I'm not thinking of, let me know!

Reply:
Ok, makes sense to me; the members of the nested type wouldn't be fulfilling member requirements declared by the header because you can't nest declarations like that in Obj-C.